### PR TITLE
first version

### PR DESCRIPTION
**Pull request by jovankomatovic**
This is just a very brief first version.
Currently, it just has a concept of how the initialization could be done.
I was focused on differences between baby blockchain init vs "normal case" init.

I guess we can discuss tomorrow (27.01) more.
After this, I will make more in-depth version.

### CODE REVIEW

File: spec/init.md [hunk 1]
```diff
@@ -0,0 +1,102 @@
+# Initialization
+
+## Introduction
+
+This document discusses how the initialization of the baby blockchain occurs.
+Namely, we consider the parent blockchain that is already operating and we describe how validators of the parent blockchain "lend" their services to the newly created baby blockchain.
+
```
Comment (josef-widder) [inline spec/init.md:7]: I think we should describe the steps from the viewpoint of a validator operator
- validator operator participates in social consensus to validate babychain => genesis file
- the validator operator uses this genesis file to 
    - start a validator node for the baby chain and starts operating
    - sign the genesis file (or a hash of it)
    - issue a transaction with the signature on the parent chain

Baby Chain
- once `>2/3` of the validators are up they may start to produce blocks.
- you run CosmosSDK with a flag `initializing` that doesn't execute any transactions except the ones needed to go through the following stages:
    - establish IBC connection / channel
    - wait for `StartBabyBlockchainPacket`
- when the `StartBabyBlockchainPacket` is received the flag `initializing` is set to false, and the normal application starts to work.

Parent chain
- when there are signatures for all validators of the baby chain `StartBabyBlockchainPacket` is sent
    - initialize staking logic for baby chain
    - disable validator set updates
- when the ACK arrives
    - enable validator set updates

Comment (josef-widder) [reply spec/init.md:7]: As far as I understand "start a validator node for the baby chain and starts operating" is done by starport?

Comment (jovankomatovic) [reply spec/init.md:7]: Sure, I will deal with this.
Can you just clarify the following questions, please?
1) Do you think we should change something that is already written or just add the points you mentioned?
2) Do you think we should talk about the points you mentioned in an "informal" way (similarly to what you wrote) or we should talk more formally (add functions, and so on)?

Thank you!

Comment (josef-widder) [reply spec/init.md:7]: I think the first points we should just describe informally (as written) as they explain expectations we have about validator operators.

The points under baby and parent chain should be formalized. Here we can be precise about function, flags, etc.

Comment (jovankomatovic) [reply spec/init.md:7]: Perfect, everything is clear!

File: spec/init.md [hunk 2]
```diff
@@ -0,0 +1,193 @@
...
+## Validator Operator Viewpoint
+
+A operator of a validator participates in a social consensus in order to validate the baby blockchain.
+The social consensus process leads to a creation of the genesis file of the baby blockchain.
+Roughly speaking, the validator operator "uses" the genesis file in two ways:
+- It starts the validator for the baby blockchain.
+- It uses the genesis file in order to issue a transaction to the parent blockchain specifying that it is indeed willing to participate as a validator on the baby blockchain (in order to fulfil the cross-chain validation concept).
+
+
```
Comment (josef-widder) [inline spec/init.md:33]: I would add a problem statement for the rest of the document:

We assume validators do what is said above.

The baby chain maintains an `initialization` variable. If

Invariants
- if both chains only perform valid transitions: if less than or equal to 2/3 signed the genesis file, the baby has initialization=true
- (no validator set changes if initialization=true)

Liveness
- if both chains only perform valid transitions and there is a correct relayer:  if more than 2/3 signed, eventually the baby sets initialized to false

### Assumption about the Babychain application

If "Initialization = true" then the application should not execute (i.e., trust) any (non-channel-establishing) transactions while in this mode.

Comment (josef-widder) [reply spec/init.md:33]: The parent chain maintains a variable `allowValidatorSetChanges` (add safety and liveness)

assumption about the parent application: you only allow validator set changes if `allowValidatorSetChanges` is true

File: spec/init.md [hunk 3]
```diff
@@ -0,0 +1,212 @@
+# Initialization
```
Comment (josef-widder) [inline spec/init.md:1]:
```suggestion
# Initialization - Draft for Discussion
```

Comment (josef-widder) [inline spec/init.md:1]:
```suggestion
# Initialization - Draft for Discussion
```

**Pull request merged**